hnn_core/tests/test_extracellular.py

```diff
@@ -195,7 +196,18 @@ def test_rec_array_calculation():
     # one electrode inside, one above the active elements of the network
     electrode_pos = [(1.5, 1.5, 1000), (1.5, 1.5, 3000)]
     net.add_electrode_array('arr1', electrode_pos)
-    _ = simulate_dipole(net, tstop=25, n_trials=1)
+    # run a single trial multiple times in this python process to ensure
+    # nothing funky happens with potential calculation callback (cf. #409)
+    run_times = list()
+    for _ in range(5):
+        start_time = time()
+        _ = simulate_dipole(net, tstop=25, n_trials=1)
+        run_times.append(time() - start_time)
+        # choice of 50% increase arbitrary (heuristic)
+        if (run_times[-1] - run_times[0]) / run_times[0] > 0.50:
+            raise RuntimeError('Time for repeated simulations increases '
+                               'greatly; check for pointer leaks in '
+                               'extracellular potential calculation callback')
```

Comment:
Instead of this test (which will be slow), another thought is to start a file call `fixes.py` that monkey-patches some of these Neuron functions/classes and then test that. For e.g., you could do:

```py
from neuron import CVode

class _CVode(CVode):

    def extra_scatter_gather_remove(self, func):
        if '.' in func.__qualname__:    # not sure if this is the best check
            raise ValueError('Only accept func attached to class')
        CVode.extra_scatter_gather_remove(self, func)
```

and then you could imagine including other Neuron fixes there. E.g., `Exp2Syn` accept floats but it's very confusing. Maybe makes sense for Neuron but we can monkey-patch like I've done [here](https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/cell.py#L510-L513)

Reply:
I'm not sure how monkey patching would do the trick here? I've tried to be very verbose in the comments, hoping that'll prevent tripping on something like this in the future.